DataConnectors/AWS-CloudTrail-Ingestion-Lambda/IngestCloudTrailEventsToSentinel.ps1

```diff
@@ -0,0 +1,345 @@
+        -sharedKey $SharedKey `
+        -date $rfc1123date `
+        -contentLength $contentLength `
+        -method $method `
+        -contentType $contentType `
+        -resource $resource
+    $uri = "https://" + $CustomerId + ".ods.opinsights.azure.com" + $resource + "?api-version=2016-04-01"
+    $headers1 = @{
+        "Authorization"        = $signature;
+        "Log-Type"             = $LogTable;
+        "x-ms-date"            = $rfc1123date;
+        "x-ms-AzureResourceId" = $resourceId;
+        "time-generated-field" = $TimeStampField;
+    }  
+    $status = $false
+    do {
+        $response = Invoke-WebRequest -Uri $uri -Method $method -ContentType $contentType -Headers $headers1 -Body $Body
+        If ($reponse.StatusCode -eq 429) {
+            $rand = get-random -minimum 10 -Maximum 80
+            start-sleep -seconds $rand 
+        }
+        else { $status = $true }
+    }until($status) 
+    Remove-variable -name Body
+    return $response.StatusCode
+    
+}
+
+
+Function Ingest-Core-Fields-Single-Table {
+	Param(
+	$coreEvents)
+	
+	$coreJson = convertto-json $coreEvents -depth 5 -Compress    
+	$Table = "$LATableName" + "_All"
+	IF (($corejson.Length) -gt 28MB) {
+		Write-Host "Log length is greater than 28 MB, splitting and sending to Log Analytics"
+		$bits = [math]::Round(($corejson.length) / 20MB) + 1
+		$TotalRecords = $coreEvents.Count
+		$RecSetSize = [math]::Round($TotalRecords / $bits) + 1
+		$start = 0
+		For ($x = 0; $x -lt $bits; $X++) {
+			IF ( ($start + $recsetsize) -gt $TotalRecords) {
+				$finish = $totalRecords
+			}
+			ELSE {
+				$finish = $start + $RecSetSize
+			}
+			$body = Convertto-Json ($coreEvents[$start..$finish]) -Depth 5 -Compress
+			$result = Invoke-LogAnalyticsData -CustomerId $workspaceId -SharedKey $workspaceKey -Body $body -LogTable $Table -TimeStampField 'eventTime' -ResourceId $ResourceID          
+			if ($result -eq 200)
+			{
+				Write-Host "CloudTrail Logs successfully ingested to LogAnalytics Workspace under Custom Logs --> Table: $Table"
+			}
+			$start = $Finish + 1
+		}
+		$null = Remove-variable -name body        
+
+	}
+	Else {
+		#$logEvents = Convertto-Json $events -depth 20 -compress
+		$result = Invoke-LogAnalyticsData -CustomerId $workspaceId -SharedKey $workspaceKey -Body $coreJson -LogTable $Table -TimeStampField 'eventTime' -ResourceId $ResourceID
+		if ($result -eq 200)
+		{
+			Write-Host "CloudTrail Logs successfully ingested to LogAnalytics Workspace under Custom Logs --> Table: $Table"
+		}
+	}
+
+	$null = remove-variable -name coreEvents
+	$null = remove-variable -name coreJson
+}
+
+
+Function Ingest-AWS-ResourceType-Multi-Tables {
+	Param(
+	$eventSources,
+	$groupEvents)
+	
+	$RecCount = 0
+	foreach ($d in $eventSources) { 
+		#$events = $groupevents[$d]
+		$eventsJson = ConvertTo-Json $groupEvents[$d] -depth 5 -Compress
+		$Table = $LATableName + '_' + $d
+		$TotalRecords = $groupEvents[$d].Count
+		$recCount += $TotalRecords
```

Comment:
extract a common function for upload to single tale and multiple table? Just need to pass in different event name and table name as param.